packages/terra-application-utility/src/utility/_UtilityMenu.jsx

```diff
@@ -0,0 +1,325 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import Button from 'terra-button';
+import ContentContainer from 'terra-content-container';
+import IconClose from 'terra-icon/lib/icon/IconClose';
+import IconLeft from 'terra-icon/lib/icon/IconLeft';
+import MenuDivider from './_UtilityMenuDivider';
+import Utils from '../_Utils';
+import MenuItem from './_UtilityMenuItem';
+import styles from './_UtilityMenu.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Key of the top level menu.
+   */
+  selectedKey: PropTypes.string.isRequired,
+  /**
+   * Indicates if the height is bound to a value.
+   */
+  isHeightBounded: PropTypes.bool,
+  /**
+   * The array of the menu items configs to be rendered.
+   */
+  menuItems: PropTypes.arrayOf((Utils.itemShape)).isRequired,
+  /**
+   * The function to trigger when a menu item is selected.
+   */
+  onChange: PropTypes.func.isRequired,
+  /**
+   * The function that closes the menu. This will be provided by the header/menu.
+   */
+  onRequestClose: PropTypes.func,
+  /**
+   * Sets the Utility variant.
+   */
+  variant: PropTypes.oneOf([Utils.VARIANTS.HEADER, Utils.VARIANTS.MENU]).isRequired,
```

Comment:
Do we need to make any of these props required if the HOC always supplies them?

Reply:
Yeah it's redundant, we can remove this